choreolib/py/choreo/version.py

```diff
@@ -17,4 +17,18 @@ def get_version():
     if proc.returncode:
         return "0.0.0"
     else:
-        return re.search(r"v([0-9]+\.[0-9]+\.[0-9]+)", proc.stdout).group(1)
+        m = re.search(
+            r"^v([0-9]+\.[0-9]+\.[0-9]+)(-([0-9]+))?(-(.*))?",
```

Comment:
.* is a greedy match, which matches too much (it matches all remaining characters, then gives back characters until the match succeeds).

The re.X argument is passed, so you can use whitespace to make the regex more legible.

Try some test strings in https://regex101.com/ to refine the regex.